File: music-service/src/main/java/com/music/service/domain/event/SongCreatedEvent.java
package com.music.service.domain.event;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SongCreatedEvent {
    private UUID songId;
    private String title;
    private String artist;
    private LocalDate releaseDate;
    private String genre;
}

File: video-enricher/Domain/Event/SongCreatedEvent.cs
using System;
using System.Text.Json.Serialization;
using VideoEnricher.Converters;

namespace VideoEnricher.Domain.Event
{
    public record SongCreatedEvent
    {
        public Guid SongId { get; init; }
        public string Title { get; init; } = string.Empty;
        public string Artist { get; init; } = string.Empty;
        
        [JsonConverter(typeof(DateArrayToStringConverter))]
        public string ReleaseDate { get; init; } = string.Empty; // Keeping as string for simplicity with ISO8601, or could be DateTime
        
        public string Genre { get; init; } = string.Empty;
    }
}
